include/odp/api/spec/packet.h

```diff
@@ -1135,6 +1135,39 @@ int odp_packet_copy_data(odp_packet_t pkt, uint32_t dst_offset,
 int odp_packet_move_data(odp_packet_t pkt, uint32_t dst_offset,
 			 uint32_t src_offset, uint32_t len);
 
+/**
+ * Set packet data to specific value
+ *
+ * Set 'len' bytes of packet data to value 'c' starting from 'offset'.  Maximum
+ * number of bytes to compare is packet data length minus the offset. Packet is
+ * not modified.
+ *
+ * @param pkt    Packet handle
+ * @param offset Byte offset into the packet
+ * @param c      Value to set
+ * @param len    Number of bytes to compare
+ */
+int odp_packet_set_data(odp_packet_t pkt, uint32_t offset,
+			uint8_t c, uint32_t len);
+
+/**
+ * Compare data with packet
+ *
+ * Compare 'len' bytes of data from the source address with the packet
+ * contentes starting from offset. Maximum number of bytes to compare is packet
+ * data length minus the offset. Packet is not modified.
+ *
+ * @param pkt    Packet handle
+ * @param offset Byte offset into the packet
+ * @param data   Data address
+ * @param len    Number of bytes to compare
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_packet_cmp_data(odp_packet_t pkt, uint32_t offset,
+			const void *data, uint32_t len);
```

Comment:
I do not need to `odp_packet_cmp_pkt()` ATM, so I will just rename this function to `odp_packet_cmp_mem()`.